### PR TITLE
Support Custom Health Metrics in the Heartbeat Tool

### LINKED ISSUE #100: Swift - Add concept of generic health metrics to Heartbeat tool
As part of the future direction of the Heartbeat tool, we want to add the concept of a user-defined list of health metrics that can bre reported on.

This can be modeled as aa array called `healthMetrics` (either in the main heartbeat collection or metadata, whichever path we think scales better).

Each entry in `healthMetrics` contains a `metricName` String describing what it is, a `isHealthy` Boolean indicating whether this metric is healthy for this device, and a `details` child map allowing users to specify a list of additional information about this metrics.

For example, a permissions health metric might look like:

```json
"healthMetrics": [
    {
        "metricName": "Device Permissions",
        "isHealthy": false,
        "details" : {
            "bluetoothAllowed": false,
            "wifiAllowed": true
        }
    }
]
```

Because this is defined in a generic way, users can define custom Ditto-related or other metrics that make sense for the specific use-case.

### PR DESCRIPTION
**Pull request by bplattenburg**
Summary:
Adds support for custom `HealthMetric`s in the Heartbeat Tool. These can be setup by implementing the `HeathMetricProvider` and passing all of your `HeathMetricProvider`s into the `HeartbeatVM` when initializing it.

Details:
- Create the concept of a `HealthMetric` which the heartbeat tool can track, used by implementing any number of `HealthMetricProvider`s and connecting them to your `DittoHeartbeatConfig`. This allows consumers to attach health metrics from `DittoSwiftTools` or create custom metrics related to application health (e.g. am I connected to the correct WiFi SSID)
- Creates new `DittoHealthMetrics` library to break out `HealthMetric` and `HealthMetricProvider` so that various tools can be extended to support the heartbeat custom health metrics but allow each tool to still be used independently.
- Add the `healthMetrics` system to `DittoHeartbeatInfo` so that data from any configured `HealthMetricProvider`s is included in the various callbacks.
- Adds a check to gather the latest information from each `HealthMetricProvider` on the configured interval, before emitting the new data to the various callbacks.
- Refactor and extend `BluetoothManager` from `DittoPermissionsHealth` tool to be a `HealthMetricProvider`
- Refactor and extend `NetworkManager` from `DittoPermissionsHealth` tool to be a `HealthMetricProvider`
- some `@available` cleanup to lower minimum versions where possible
- Minor cleanup of `Package.swift` to remove redundant paths that match the default SPM behavior
- SPM versioning alignment between `DittoSwiftTools` and `DittoToolsApp`

Closes #100

### CODE REVIEW

File: DittoToolsApp/DittoToolsApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -784,7 +784,7 @@
 			repositoryURL = "https://github.com/getditto/DittoSwiftPackage";
 			requirement = {
 				kind = upToNextMajorVersion;
-				minimumVersion = 4.5.3;
+				minimumVersion = 4.5.0;
```
Comment (bplattenburg) [inline DittoToolsApp/DittoToolsApp.xcodeproj/project.pbxproj:787]: I aligned this to the version specified in the `Package.swift` of the `DittoSwiftTools` package, but longer term I wonder if we want to explicitly include this dependency here or let it be pulled transitively by the fact that we include the local `DittoSwiftTools` package which has the dependency already.

Pros of including it this way:
- Allows for compatibility testing with specific versions

Cons
- An extra spot to maintain the version number in sync

**Comment by bplattenburg** [comment]
I'm leaving this as a draft right now because I'm unable to test it yet due to local issues running the `DittoToolsApp`.

**Review by rajramsaroop — APPROVED** [review]

**Comment by bplattenburg** [comment]
@texasRanger09 Updated the README to match those changes.

I also broke out `HealthMetric` and `HealthMetricProvider` into a new product/target called `DittoToolsSharedModels` to be able to extend existing tools to conform to `HealthMetricProvider` and provided a sample implemenation for both `BluetoothManager` and `NetworkManager` in the Permissions Health tool (WIP)

**Comment by bplattenburg** [comment]
> I'm leaving this as a draft right now because I'm unable to test it yet due to local issues running the `DittoToolsApp`.

No longer a draft as I can now test this out by linking to the pushed branch instead of the local reference. Logged https://github.com/getditto/DittoSwiftTools/issues/106 to try to resolve that issue later

**Comment by texasRanger09** [comment]
LGTM

**Review by texasRanger09 — APPROVED** [review]

**Pull request merged**